File: Gym_App/src/main/java/com/backend/App_gym/entity/Usuario.java
package com.backend.App_gym.entity;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "user")
public class Usuario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_user")
    private Long id_user;
    
    @Column(nullable = false, unique = true)
    private String username;
    
    @Column(nullable = false)
    private String password;
    
    @Column(nullable = false)
    private String name;
    
    @Column(nullable = false)
    private String lastname;
    
    @Column(nullable = false)
    private LocalDate date;
    
    @Column(nullable = false, unique = true)
    private String email;
    
    @Column(nullable = false)
    private String sex;
    
    @OneToOne(mappedBy = "usuario", cascade = CascadeType.ALL)
    private UserData userData;

    // Getters y Setters
    public Long getId_user() {
        return id_user;
    }

    public void setId_user(Long id_user) {
        this.id_user = id_user;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLastname() {
        return lastname;
    }

    public void setLastname(String lastname) {
        this.lastname = lastname;
    }

    public LocalDate getDate() {
        return date;
    }

    public void setDate(LocalDate date) {
        this.date = date;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getSex() {
        return sex;
    }

    public void setSex(String sex) {
        this.sex = sex;
    }

    public UserData getUserData() {
        return userData;
    }

    public void setUserData(UserData userData) {
        this.userData = userData;
    }
} 

File: Gym_App/app/src/main/java/com/cristian/appgym/model/Usuario.kt
package com.cristian.appgym.model

import com.google.gson.annotations.SerializedName

data class Usuario(
    @SerializedName("id_user")
    val id_user: Long? = null,
    
    @SerializedName("userId")
    val userId: Long? = null,
    
    @SerializedName("username")
    val username: String,
    
    @SerializedName("password")
    val password: String,
    
    @SerializedName("name")
    val name: String,
    
    @SerializedName("lastname")
    val lastname: String,
    
    @SerializedName("email")
    val email: String,
    
    @SerializedName("date")
    val date: String,
    
    @SerializedName("sex")
    val sex: String,
    
    @SerializedName("userData")
    val userData: UserData? = null
)